doc/kdump/kdump_Remote_SSH_HLD.md

```diff
@@ -0,0 +1,228 @@
+
+## kdump Remote Architechture
+![alt text](./images/kdump-ssh.drawio.png)
+
+## Functional Description
+
+### Design Changes
+
+The SONiC kernel core dump remote functionality can be divided into two categories:
+
+1. Kernel core-dump generation service
+2. Storing Kernel core-dump files remotely
+
+Current SONiC lacks remote kernel dump functionality. To add this feature, consider enabling kdump for remote storage.
+
+We are suggesting modifying the exisitng SONiC configuration as following sonic-buildimage files.
+ 
+1. build_debian.sh (Addition)
+
+    - Required for kdump_remote_ssh_dump: InItialize network interfaces and enable DHCP upon kernel crash. 
+
+2. files/scripts/network_setup.sh (New Addition)
+    - A script to initialize the network interfaces and enable DHCP on them.
+3. files/script/network_setup.hook (New Addition)
+4. Added a daemon into files/image_config/kdump-feature/kdump-service.py (New Addition).
+
+## Configuration and Management
+This section describes all types of configuration and management related design. Example sub-sections for "CLI" and "Config DB" are given below.
+
+
+
+### CONFIG_DB Enhancements
+ New attributes will be introduced to "KDUMP" table in ConfigDB for maintaining remote kdump configurations. Below is the schema for this table.
+
+```
+    KDUMP_TABLE:{{config}}
+     "enabled"    :{{"false"|"true"}}
+     "memory"     :{{string}}
+     "num_dumps"  :{{number}}
+     "remote"     :{{"false"|"true"}}
+     "ssh_string" :{{string}}
+     "ssh_path" :{{string}}
+```
+
+### SAI API 
+
+No SAI API change or addition is needed for this HLD. 
+
+
+### CLI/YANG model Enhancements 
+### CLI Enhancements
+
+#### CLI Configuration Commands
+New SONiC CLI commands are introduced to configure remote kdump feature.
+
+```
+admin@sonic: sudo config kdump remote enable 
+```
+This commans will configure remote kdump feature in SONiC.
+
+```
+admin@sonic: sudo config kdump remote add ssh_string username@serverip
+```
+
+```
+admin@sonic: sudo config kdump remote add ssh_path /path to ssh private key/
+```
+
+```
+admin@sonic: sudo config kdump remove ssh_sting
+```
+
+```
+admin@sonic: sudo config kdump remove ssh_path
+```
+
+```
+admin@sonic: sudo config kdump remote disable
+```
+This command disables the remote kdump feature.
+#### CLI Show Commands
+An existing SONiC CLI command is used to display the current remote kdump feature configuraitons.
+
+```
+admin@sonic: show kdump config
+```
+
+Example output from the above command:
+```
+admin@sonic:~$ show kdump config
+Kdump administrative mode: Enabled
+Kdump operational mode: Ready
+Kdump memory reservation: 512
+Maximum number of Kdump files: 3
+remote: true
+ssh_string: username@serverip
+ssh_path: /path to ssh_private key/
+```
+
+### YANG Enhancements
+
+```
+leaf remote {
+   description "SSH Remote Config";
+     type boolean;
+        }
+
+leaf ssh_stirng {
+    type string;
+}
+
+leaf ssh_path {
```

Comment:
Please add a description explaining the format for all fields.

Reply:
Updated